crates/bevy_ecs/src/reflect/from_world.rs

```diff
@@ -0,0 +1,84 @@
+//! Definitions for [`FromWorld`] reflection.
```

Comment:
This could use a sentence or two of docs about why this is useful.

Reply:
Note that this is a private module. I tried to expand the documentation a bit anyway, though I'm not fully convinced. I also expanded a bit the documentation of the `bundle` and `component` modules to keep the same "structure".

Related: I just noticed that the `component` module has some kinda extensive documentation, but unfortunately it is not visible to users due to the module being private.